### ISSUE TITLE
apply_async's "shadow" parameter not working?

### ISSUE THREAD

**Issue by jmdacruz** [celery/celery#4347]
If I'm reading correctly, [apply_async](http://docs.celeryproject.org/en/latest/reference/celery.app.task.html#celery.app.task.Task.apply_async) has an argument called `shadow`, which overrides the task name on the logs and monitoring. This doesn't seem to be working, or I'm doing something wrong (quite possibly)

## Checklist

- [X] I have included the output of ``celery -A proj report`` in the issue.
      (if you are not able to do this, then at least specify the Celery
       version affected).
- [x] I have verified that the issue exists against the `master` branch of Celery.

### output of ``celery -A proj report``

```
software -> celery:4.1.0 (latentcall) kombu:4.1.0 py:2.7.12
            billiard:3.5.0.3 redis:2.10.6
platform -> system:Linux arch:64bit, ELF imp:CPython
loader   -> celery.loaders.app.AppLoader
settings -> transport:redis results:redis://localhost:6379/4

broker_url: u'redis://localhost:6379/4'
result_backend: u'redis://localhost:6379/4'
```

## Steps to reproduce

Create an application based on the example in [First Steps With Celery](http://docs.celeryproject.org/en/latest/getting-started/first-steps-with-celery.html)

* `tasks.py`

```python
from celery import Celery

app = Celery('tasks', broker='redis://localhost:6379/4', backend='redis://localhost:6379/4')

@app.task
def add(x, y):
    return x + y
```

* `main.py`

```python
from tasks import add

if __name__ == "__main__":

    t1 = add.apply_async(args=(2,2), shadow="foo")
    t2 = add.apply_async(args=(3,3), shadow="bar")

    r1 = t1.get()
    r2 = t2.get()
    print("R1 = {}, R2 = {}".format(r1, r2))
```

Execute the worker:

```
celery -A tasks worker --loglevel=info
```

Execute flower:

```
flower -A tasks
```

## Expected behavior

Flower's "tasks" dashboard shows the task names as "foo" and "bar".

## Actual behavior

Flower's "tasks" dashboard shows the task names as "tasks.add" in both cases. The worker's log also shows the original task name without modification:

```
[2017-10-26 20:29:22,153: INFO/MainProcess] Received task: tasks.add[351ff35e-a834-4ccf-aa25-28239367024e]
[2017-10-26 20:29:22,155: INFO/MainProcess] Received task: tasks.add[f6a411fe-d3c8-4c8a-9604-8b892d901088]
[2017-10-26 20:29:22,164: INFO/ForkPoolWorker-8] Task tasks.add[351ff35e-a834-4ccf-aa25-28239367024e] succeeded in 0.00483218299632s: 4
[2017-10-26 20:29:22,164: INFO/ForkPoolWorker-3] Task tasks.add[f6a411fe-d3c8-4c8a-9604-8b892d901088] succeeded in 0.00474107099581s: 6
```

**Comment by jmdacruz** [comment]
Effectively, it seems that the code to add the `shadow` property to the task message is not in master either: https://github.com/celery/celery/blob/master/celery/app/amqp.py#L354. It seems that the `as_task_v2` method is ignoring the `shadow` parameter when creating the task message

**Comment by jmdacruz** [comment]
I can confirm that PR #4350 fixes this issue (in monitoring tools such as flower):

![image](https://user-images.githubusercontent.com/1178370/32139726-0e6a5c92-bc07-11e7-9349-ac28e109d45a.png)

Logs are also "partially" fixed (the "received" log entries show the shadow name, but the "succeeded" log entries show the real name):

```
[2017-10-28 17:39:17,760: INFO/MainProcess] Received task: foo[fcfbb11e-f565-4630-8aa6-eac3434226c1]
[2017-10-28 17:39:17,762: INFO/MainProcess] Received task: bar[f1ac07ad-bc60-4ec2-a6b5-b246dec4ce01]
[2017-10-28 17:39:17,771: INFO/ForkPoolWorker-4] Task tasks.add[fcfbb11e-f565-4630-8aa6-eac3434226c1] succeeded in 0.00656166800763s: 4
[2017-10-28 17:39:17,771: INFO/ForkPoolWorker-5] Task tasks.add[f1ac07ad-bc60-4ec2-a6b5-b246dec4ce01] succeeded in 0.00663297998835s: 6
```

**Comment by jmdacruz** [comment]
Closed by #4350
